// apollo-server/dist/index.js
"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const express_1 = __importDefault(require("express"));
const apollo_server_express_1 = require("apollo-server-express");
const apollo_server_core_1 = require("apollo-server-core");
__export(require("./exports"));
class ApolloServer extends apollo_server_core_1.ApolloServerBase {
    async listen(opts = {}) {
        if (!this.getHttp) {
            const app = express_1.default();
            await apollo_server_express_1.registerServer({
                app,
                path: '/',
                server: this,
                bodyParserConfig: { limit: '50mb' },
                cors: {
                    origin: '*',
                },
            });
        }
        return super.listen(opts);
    }
}
exports.ApolloServer = ApolloServer;
